src/darwin/Framework/CHIP/EndpointDescription/MTRAttributeDescription.h

```diff
@@ -0,0 +1,57 @@
+/**
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRDefines.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * A representation of an attribute implemented on a server cluster by an
+ * MTRDeviceController.  An attribute has an identifier and a value, and may or
+ * may not be writable.
+ */
+MTR_NEWLY_AVAILABLE
+@interface MTRAttributeDescription : NSObject <NSCopying>
```

Comment:
I guess I'm missing the context here of how this is meant to be used overall, but it seems a little surprising that the attribute value is fixed as part of the "description". If the server code needs to update the value, does the cluster need to be updated with a new attribute description? Getting callbacks for writes also seems like a useful thing.

Reply:
Context: this is providing the initial value of the attribute.  There will also be API on the controller to update the value of the attribute.  The update does not go through the *Description bits.

Callbacks for writes will happen once we support writes.  For now we don't need that support, so I am not actually implementing it (and initially will reject attempts to add writable attributes....)

I guess I could remove the "init as writable" bit until that writable support happens.

Reply:
I removed the writable attribute bits.